framework_tvb/tvb/adapters/uploaders/bids_importer.py

```diff
@@ -0,0 +1,288 @@
+from tvb.datatypes.time_series import TimeSeriesRegion
+
+
+class BIDSImporterModel(UploaderViewModel):
+    uploaded = Str(
+        label='BIDS dataset (zip)'
+    )
+
+
+class BIDSImporterForm(ABCUploaderForm):
+
+    def __init__(self):
+        super(BIDSImporterForm, self).__init__()
+
+        self.uploaded = TraitUploadField(BIDSImporterModel.uploaded, '.zip', 'uploaded')
+
+    @staticmethod
+    def get_view_model():
+        return BIDSImporterModel
+
+    @staticmethod
+    def get_upload_information():
+        return {
+            'uploaded': '.zip'
+        }
+
+
+class BIDSImporter(ABCUploader):
+
+    _ui_name = "BIDS Importer"
+    _ui_subsection = "bids_importer"
+    _ui_description = "Import a dataset in BIDS format"
+
+    SUBJECT_PREFIX = "sub"
+
+    NET_TOKEN = "net"
+    COORDS_TOKEN = "coord"
+    SPATIAL_TOKEN = "spatial"
+    TS_TOKEN = "ts"
+
+    TSV_EXTENSION = ".tsv"
+    JSON_EXTENSION = ".json"
+
+    WEIGHTS_FILE = "weights" + TSV_EXTENSION
+    WEIGHTS_JSON_FILE = "weights" + JSON_EXTENSION
+    DISTANCES_FILE = "distances" + TSV_EXTENSION
+
+    VERTICES_FILE = "vertices" + TSV_EXTENSION
+    NORMALS_FILE = "normals" + TSV_EXTENSION
+    TRIANGLES_FILE = "faces" + TSV_EXTENSION
+
+    COORDS_ROWS_KEY = "CoordsRows"
+
+    def get_form_class(self):
+        return BIDSImporterForm
+
+    def get_output(self):
+        return [VolumeIndex, StructuralMRIIndex, TimeSeriesVolumeIndex, RegionVolumeMappingIndex]
+
+    def launch(self, view_model):
+        """
+        Import a dataset in BIDS format
+        """
+
+        if view_model.uploaded is None:
+            raise LaunchException("Please select ZIP file which contains data to import")
+
+        files = self.storage_interface.unpack_zip(view_model.uploaded, self.get_storage_path())
+        subject_folders = []
+
+        # First we find subject parent folders
+        for file_name in files:
+            if os.path.basename(file_name).startswith(self.SUBJECT_PREFIX) and os.path.isdir(file_name):
+                subject_folders.append(file_name)
+
+        connectivity = None
+        ts_dict = None
+
+        for subject_folder in subject_folders:
+            net_folder = os.path.join(subject_folder, self.NET_TOKEN)
+            if os.path.exists(net_folder):
+                connectivity = self.__build_connectivity(net_folder)
+
+            coords_folder = os.path.join(subject_folder, self.COORDS_TOKEN)
+            if os.path.exists(coords_folder):
+                self.__build_surface(coords_folder)
+
+            ts_folder = os.path.join(subject_folder, self.TS_TOKEN)
+            if os.path.exists(ts_folder):
+                ts_dict = self.__build_time_series(ts_folder, connectivity)
+
+            spatial_folder = os.path.join(subject_folder, self.SPATIAL_TOKEN)
+            if os.path.exists(spatial_folder):
+                self.__build_functional_connectivity(spatial_folder, ts_dict)
+
+    def __build_connectivity(self, net_folder):
```

Comment:
The methods, build_connectivity, build_surface_build_time_series, build_functional_connectivity can be put into their own python files having relevant file names which define their functionalities.